packages/@o3r/design/src/core/design-token/parsers/design-token.parser.ts

```diff
@@ -0,0 +1,172 @@
+const getReferences = (cssRawValue: string) => Array.from(cssRawValue.matchAll(tokenReferenceRegExp)).map(([,tokenRef]) => tokenRef);
+
+// eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
+const renderCssTypeStrokeStyleValue = (value: DesignTokenTypeStrokeStyleValue | string) => isTokenTypeStrokeStyleValueComplex(value) ? `${value.lineCap} ${value.dashArray.join(' ')}` : value;
+
+const getCssRawValue = (variableSet: DesignTokenVariableSet, {node, getType}: DesignTokenVariableStructure) => {
+  const nodeType = getType(variableSet, false);
+  if (!nodeType && node.$value) {
+    return typeof node.$value.toString !== undefined ? (node.$value as any).toString() : JSON.stringify(node.$value);
+  }
+  const checkNode = {
+    ...node,
+    $type: node.$type || nodeType
+  } as typeof node;
+
+  // TODO in the following code, `typeof checkNode.$value === 'string' ? checkNode.$value :` is defined to please Jest TS compilation. It should be removed when supported
+  switch (checkNode.$type) {
+    case 'color':
+    case 'number':
+    case 'duration':
+    case 'fontWeight':
+    case 'fontFamily':
+    case 'dimension': {
+      return checkNode.$value.toString();
+    }
+    case 'strokeStyle': {
+      return renderCssTypeStrokeStyleValue(checkNode.$value);
+    }
+    case 'cubicBezier': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        checkNode.$value.join(', ');
+    }
+    case 'border': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        `${checkNode.$value.width} ${renderCssTypeStrokeStyleValue(checkNode.$value.style)} ${checkNode.$value.color}`;
+    }
+    case 'gradient': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        `linear-gradient(0deg, ${checkNode.$value.map(({color, position}) => `${color} ${position}`).join(', ')})`;
+    }
+    case 'shadow': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        `${checkNode.$value.offsetX} ${checkNode.$value.offsetY} ${checkNode.$value.blur}  ${checkNode.$value.spread} ${checkNode.$value.color}`;
+    }
+    case 'transition': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        typeof checkNode.$value.timingFunction === 'string' ? checkNode.$value.timingFunction : checkNode.$value.timingFunction.join(' ') +
+          ` ${checkNode.$value.duration} ${checkNode.$value.delay}`;
+    }
+    case 'typography': {
+      return typeof checkNode.$value === 'string' ? checkNode.$value :
+        `${checkNode.$value.fontWeight} ${checkNode.$value.fontFamily} ${checkNode.$value.fontSize} ${checkNode.$value.letterSpacing} ${checkNode.$value.lineHeight}`;
+    }
+    default: {
+      throw new Error(`Not supported type ${(checkNode as any).$type || 'unknown'} (value: ${(checkNode as any).$value || 'unknown'})`);
```

Comment:
```suggestion
      throw new O3rCliError(`Not supported type ${(checkNode as any).$type || 'unknown'} (value: ${(checkNode as any).$value || 'unknown'})`);
```

Reply:
Same reponse as for the CLI, I do not want to add a dep to @o3r/schematics in the src code